pkg/cmd/alias/imports/imports_test.go

```diff
@@ -0,0 +1,279 @@
+		},
+		{
+			name:  "override aliases",
+			isTTY: true,
+			flags: []string{"--clobber"},
+			fileContents: heredoc.Doc(`
+                co: pr checkout -R cool/repo
+                igrep: '!gh issue list --label="$1" | grep "$2"'
+            `),
+			initialConfig: heredoc.Doc(`
+                aliases:
+                    co: pr checkout
+                editor: vim
+            `),
+			expectedConfig: heredoc.Doc(`
+                aliases:
+                    co: pr checkout -R cool/repo
+                    igrep: '!gh issue list --label="$1" | grep "$2"'
+                editor: vim
+            `),
+			expectedErrLines: []string{
+				"Importing aliases from file",
+				"Changed alias co",
+				"Added alias igrep",
+			},
+			expectedOutLines: []string{},
+		},
+		{
+			name:  "alias is a gh command",
+			isTTY: true,
+			flags: []string{},
+			fileContents: heredoc.Doc(`
+                pr: pr checkout
+                issue: issue list
+                api: api graphql
+            `),
+			expectedErrLines: []string{"Could not import alias (pr|checkout|list): already a gh command"},
+			expectedOutLines: []string{},
+		},
+		{
+			name:  "invalid expansion",
+			isTTY: true,
+			flags: []string{},
+			fileContents: heredoc.Doc(`
+                alias1:
+                alias2: ps checkout
+            `),
+			expectedErrLines: []string{"Could not import alias alias[0-9]: expansion does not correspond to a gh command"},
+			expectedOutLines: []string{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cli := "-"
+			if tt.stdin == "" {
+				tmpFile := filepath.Join(t.TempDir(), "aliases")
+				err := os.WriteFile(tmpFile, []byte(tt.fileContents), 0600)
+				if err != nil {
+					t.Fatalf("could not write test file %v", err)
+				}
+				cli = tmpFile
+			}
+			cli = strings.Join(append(tt.flags, cli), " ")
+
+			readConfigs := config.StubWriteConfig(t)
+
+			cfg := config.NewFromString(tt.initialConfig)
+
+			output, err := runCommand(cfg, tt.isTTY, cli, tt.stdin)
+			require.NoError(t, err)
+
+			mainBuf := bytes.Buffer{}
+			readConfigs(&mainBuf, io.Discard)
+
+			//nolint:staticcheck // prefer exact matchers over ExpectLines
+			test.ExpectLines(t, output.Stderr(), tt.expectedErrLines...)
+			//nolint:staticcheck // prefer exact matchers over ExpectLines
+			test.ExpectLines(t, output.String(), tt.expectedOutLines...)
+
+			assert.Equal(t, tt.expectedConfig, mainBuf.String())
+		})
+	}
+}
+
+func TestAliasImports_no_filename(t *testing.T) {
+	cfg := config.NewFromString("")
+	_, err := runCommand(cfg, true, "", "")
+	require.EqualError(t, err, "no filename passed and nothing on STDIN")
+}
+
+func TestAliasImports_multiple_filenames(t *testing.T) {
+	cfg := config.NewFromString("")
+	_, err := runCommand(cfg, true, "aliases1 aliases2", "")
+	require.EqualError(t, err, "too many arguments")
+}
+
+func runCommand(cfg config.Config, isTTY bool, cli, in string) (*test.CmdOut, error) {
```

Comment:
We have generally moved away from `runCommand` style testing in favor of testing the setup (`NewCmdImports`) and run (`importsRun`) functions separately. Would you mind refactoring these tests to fit the new pattern? You can find a good example of the pattern in `gist/create/create_test.go`.